File: app/core/src/main/java/com/fsck/k9/notification/NotificationActionCreator.java
package com.fsck.k9.notification;


import java.util.List;

import android.app.PendingIntent;
import android.content.Context;

import com.fsck.k9.Account;
import com.fsck.k9.controller.MessageReference;


public interface NotificationActionCreator {
    PendingIntent createViewMessagePendingIntent(MessageReference messageReference, int notificationId);

    PendingIntent createViewFolderPendingIntent(Account account, long folderId, int notificationId);

    PendingIntent createViewMessagesPendingIntent(Account account, List<MessageReference> messageReferences,
            int notificationId);

    PendingIntent createViewFolderListPendingIntent(Account account, int notificationId);

    PendingIntent createDismissAllMessagesPendingIntent(Account account, int notificationId);

    PendingIntent createDismissMessagePendingIntent(Context context, MessageReference messageReference,
            int notificationId);

    PendingIntent createReplyPendingIntent(MessageReference messageReference, int notificationId);

    PendingIntent createMarkMessageAsReadPendingIntent(MessageReference messageReference, int notificationId);

    PendingIntent createMarkAllAsReadPendingIntent(Account account, List<MessageReference> messageReferences,
            int notificationId);

    PendingIntent createMuteSenderPendingIntent(MessageReference messageReference, int notificationId);

    PendingIntent getEditIncomingServerSettingsIntent(Account account);

    PendingIntent getEditOutgoingServerSettingsIntent(Account account);

    PendingIntent createDeleteMessagePendingIntent(MessageReference messageReference, int notificationId);

    PendingIntent createDeleteAllPendingIntent(Account account, List<MessageReference> messageReferences,
            int notificationId);

    PendingIntent createArchiveMessagePendingIntent(MessageReference messageReference, int notificationId);

    PendingIntent createArchiveAllPendingIntent(Account account, List<MessageReference> messageReferences,
            int notificationId);

    PendingIntent createMarkMessageAsSpamPendingIntent(MessageReference messageReference, int notificationId);
}

File: app/core/src/main/java/com/fsck/k9/notification/NotificationActionCreator.kt
package com.fsck.k9.notification

import android.app.PendingIntent
import com.fsck.k9.Account
import com.fsck.k9.controller.MessageReference

interface NotificationActionCreator {
    fun createViewMessagePendingIntent(messageReference: MessageReference, notificationId: Int): PendingIntent

    fun createViewFolderPendingIntent(account: Account, folderId: Long, notificationId: Int): PendingIntent

    fun createViewMessagesPendingIntent(
        account: Account,
        messageReferences: List<MessageReference>,
        notificationId: Int
    ): PendingIntent

    fun createViewFolderListPendingIntent(account: Account, notificationId: Int): PendingIntent

    fun createDismissAllMessagesPendingIntent(account: Account, notificationId: Int): PendingIntent

    fun createDismissMessagePendingIntent(
        messageReference: MessageReference,
        notificationId: Int
    ): PendingIntent

    fun createReplyPendingIntent(messageReference: MessageReference, notificationId: Int): PendingIntent

    fun createMarkMessageAsReadPendingIntent(messageReference: MessageReference, notificationId: Int): PendingIntent

    fun createMarkAllAsReadPendingIntent(
        account: Account,
        messageReferences: List<MessageReference>,
        notificationId: Int
    ): PendingIntent

    fun getEditIncomingServerSettingsIntent(account: Account): PendingIntent

    fun getEditOutgoingServerSettingsIntent(account: Account): PendingIntent

    fun createDeleteMessagePendingIntent(messageReference: MessageReference, notificationId: Int): PendingIntent

    fun createDeleteAllPendingIntent(
        account: Account,
        messageReferences: List<MessageReference>,
        notificationId: Int
    ): PendingIntent

    fun createArchiveMessagePendingIntent(messageReference: MessageReference, notificationId: Int): PendingIntent

    fun createArchiveAllPendingIntent(
        account: Account,
        messageReferences: List<MessageReference>,
        notificationId: Int
    ): PendingIntent

    fun createMarkMessageAsSpamPendingIntent(messageReference: MessageReference, notificationId: Int): PendingIntent
}
